app/api.py

```diff
@@ -38,3 +44,194 @@ def main_page(request: Request):
 @app.get("/robots.txt", response_class=PlainTextResponse)
 def robots_txt():
     return """User-agent: *\nDisallow: /"""
+
+
+# CRUD Flags
+
+
+class TicketCreate(BaseModel):
+    barcode: str = Field(..., description="Barcode of the product")
+    type: str = Field(..., description="Type of the issue")
+    url: str = Field(..., description="URL of the product")
+    status: str = Field(..., description="Status of the ticket")
+    image_id: str = Field(..., description="Image ID of the product")
+    flavour: Flavor = Field(..., description="Flavour of the product")
+    created_at: datetime = Field(default_factory=datetime.utcnow)
+
+
+class Ticket(TicketCreate):
+    id: int = Field(..., description="ID of the ticket")
+
+
+class FlagCreate(BaseModel):
+    barcode: str = Field(..., description="Barcode of the product")
+    type: str = Field(..., description="Type of the issue")
+    url: str = Field(..., description="URL of the product")
+    user_id: str = Field(..., description="User ID of the flagger")
+    device_id: str = Field(..., description="Device ID of the flagger")
+    source: str = Field(..., description="Source of the flag")
+    confidence: float = Field(..., description="Confidence of the flag")
+    image_id: str = Field(..., description="Image ID of the product")
+    flavour: Flavor = Field(..., description="Flavour of the product")
+    reason: str = Field(..., description="Reason of the flag")
+    comment: str = Field(..., description="Comment of the flag")
+    created_at: datetime = Field(default_factory=datetime.utcnow)
+
+
+class Flag(FlagCreate):
+    id: int = Field(..., description="ID of the flag")
+    ticket_id: int = Field(..., description="ID of the ticket associated with the flag")
+
+
+# Create a flag (one to one relationship)
+@app.post("/flags")
+def create_flag(flag: FlagCreate) -> Flag:
+    with db:
+        try:
+            # Search for existing ticket
+            # With the same barcode, url, type and flavour
+            ticket = TicketModel.get_or_none(
+                TicketModel.barcode == flag.barcode,
+                TicketModel.url == flag.url,
+                TicketModel.type == flag.type,
+                TicketModel.flavour == flag.flavour,
+            )
+            # If no ticket found, create a new one
+            if ticket is None:
+                newTicket = TicketCreate(
+                    barcode=flag.barcode,
+                    url=flag.url,
+                    type=flag.type,
+                    flavour=flag.flavour,
+                    status="open",
+                    image_id=flag.image_id,
+                )
+                ticket = _create_ticket(newTicket)
+            new_flag = FlagModel.create(**flag.model_dump())
+            # Associate the flag with the ticket
+            new_flag.ticket_id = ticket.id
+            new_flag.save()
+            return new_flag
+        except Exception as error:
```

Comment:
This is better done using a middleware, instead of having a try/catch on each route